### PR TITLE
Painless script implementation (PEG edition)

### PR DESCRIPTION
**Pull request by nablaone**
This PR adds a rudimentary implementation of the Painless script. This is the first usage of the "PEG" parser.

### CODE REVIEW

File: quesma/painful/painless.peg [hunk 1]
```diff
@@ -0,0 +1,120 @@
+{
+package painful
+}
+
+Expr = expr:OpExpr / expr:MethodCall / expr:Accessor / expr:Doc / expr:Emit   {
+    return expr,nil
```
Comment (jakozaur) [inline quesma/painful/painless.peg:6]:
```suggestion
    return expr, nil
```

File: quesma/painful/painless.peg [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+    exprVal ,err := ExpectExpr(expr)
+    if err != nil {
+        return nil, err
+    }
+
+    return &EmitExpr{Expr: exprVal}, nil
+}
+
+
+Doc = "doc" "[" '\'' key:Identifier '\'' "]" {
```
Comment (jakozaur) [inline quesma/painful/painless.peg:20]: Does `doc['xyz']` make sense, maybe it should always be `doc['xyz'].value`

File: quesma/painful/painless.peg [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+
+    keyVal,err := ExpectString(key)
```
Comment (jakozaur) [inline quesma/painful/painless.peg:22]: In some many places I would add space after comma.

**Review by jakozaur — COMMENTED** [review]
Nice progress.

File: quesma/painful/painless.peg (hunk 1 above)
Comment (nablaone) [reply quesma/painful/painless.peg]: Done

File: quesma/quesma/router.go [hunk 4]
```diff
@@ -58,6 +59,41 @@ func ConfigureRouter(cfg *config.QuesmaConfiguration, sr schema.Registry, lm *cl
 	// So, if you add multiple handlers with the same path, the first one will be used, the rest will be redirected to the elastic cluster.
 	// This is current limitation of the router.
 
+	router.Register(routes.ExecutePainlessScriptPath, and(method("POST"), matchAgainstIndexNameInScriptRequestBody(tableResolver)), func(ctx context.Context, req *mux.Request) (*mux.Result, error) {
```
Comment (pdelewski) [inline quesma/quesma/router.go:62]: This needs to be added to `routerV2.go` as well

Comment (nablaone) [reply quesma/quesma/router.go:62]: Done.

File: quesma/v2/core/routes/paths.go [hunk 5]
```diff
@@ -7,25 +7,26 @@ import (
...
+	EQLSearch                 = "/:index/_eql/search"
+	ResolveIndexPath          = "/_resolve/index/:index"
+	ClusterHealthPath         = "/_cluster/health"
+	BulkPath                  = "/_bulk"
+	AsyncSearchIdPrefix       = "/_async_search/"
+	AsyncSearchIdPath         = "/_async_search/:id"
+	AsyncSearchStatusPath     = "/_async_search/status/:id"
+	KibanaInternalPrefix      = "/.kibana_"
+	IndexPath                 = "/:index"
+	ExecutePainlessScriptPath = "/_scripts/painless/_execute"
```
Comment (pdelewski) [inline quesma/v2/core/routes/paths.go:29]: How full workflow looks like? Is this first request from client to send some context information?

Comment (nablaone) [reply quesma/v2/core/routes/paths.go:29]: When editing a script kibana provides a playground.   Kibana sends the script and sample data and renders the result. 

Screenshot 2024-12-06 at 11 59 31

**Comment by pdelewski** [comment]
LGTM :)

**Review by pdelewski — APPROVED** [review]

**Pull request merged**